scripts/kconfig/menuconfig.py

```diff
@@ -2229,6 +2245,36 @@ def _safe_move(win, *args):
     except curses.error:
         pass
 
+def _convert_c_lc_ctype_to_utf8():
+    # See _CONVERT_C_LOCALE_TO_UTF8
+
+    if _IS_WINDOWS:
+        # Windows rarely has issues here, and the PEP 538 implementation avoids
+        # changing the locale on it. None of the UTF-8 locales below were
+        # supported from some quick testing either. Play it safe.
+        return
+
+    def _try_set_locale(loc):
+        try:
+            locale.setlocale(locale.LC_CTYPE, loc)
+            return True
+        except locale.Error:
+            return False
+
+    # Is LC_CTYPE set to the C locale?
+    if locale.setlocale(locale.LC_CTYPE, None) == "C":
+        # This list was taken from the PEP 538 implementation in the CPython
+        # code, in Python/pylifecycle.c
+        for loc in "C.UTF-8", "C.utf8", "UTF-8":
+            if _try_set_locale(loc):
+                print("Note: Your environment is configured to use ASCII. To "
+                      "avoid Unicode issues, LC_CTYPE was changed from the "
+                      "C locale to the {} locale.".format(loc))
+                break
+
+# Are we running on Windows?
+_IS_WINDOWS = (platform.system() == "Windows")
```

Comment:
Why not use `sys.platform()` instead, to be aligned with `pip`?

Reply:
I was planning to use `platform_system` for `pip` as well, but googling "pip platform_system" turns up some bug reports that show that support was added to `pip` pretty recently (though it's been in Python itself for a long time). `sys_platform` is probably safer for `pip`.

`platform.system()` seems to be the "recommended" interface these days. It can do a bunch of dynamic querying and stuff too. Won't really matter here though, but at least "Windows" is less confusing than "win32". :)

Reply:
Understood, happy with the explanation